nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/excel/ExcelRecordReader.java

```diff
@@ -0,0 +1,221 @@
+import org.apache.nifi.serialization.record.DataType;
+import org.apache.nifi.serialization.record.MapRecord;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.serialization.record.RecordField;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.serialization.record.util.DataTypeUtils;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.DateUtil;
+import org.apache.poi.ss.usermodel.Row;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+public class ExcelRecordReader implements RecordReader {
+    private final RowIterator rowIterator;
+    private ComponentLog logger;
+    private final RecordSchema schema;
+    private final List<String> desiredSheets;
+    private final Supplier<DateFormat> LAZY_DATE_FORMAT;
+    private final Supplier<DateFormat> LAZY_TIME_FORMAT;
+    private final Supplier<DateFormat> LAZY_TIMESTAMP_FORMAT;
+    private final String dateFormat;
+    private final String timeFormat;
+    private final String timestampFormat;
+
+    public ExcelRecordReader(ExcelRecordReaderArgs args) throws MalformedRecordException {
+        this.logger = args.getLogger();
+        this.schema = args.getSchema();
+        this.desiredSheets = new ArrayList<>();
+        if (args.getDesiredSheets() != null && args.getDesiredSheets().length() > 0) {
+            IntStream.range(0, args.getDesiredSheets().length())
+                    .forEach(index -> this.desiredSheets.add(args.getDesiredSheets().get(index)));
+        }
+
+        if (isEmpty(args.getDateFormat())) {
+            this.dateFormat = null;
+            LAZY_DATE_FORMAT = null;
+        } else {
+            this.dateFormat = args.getDateFormat();
+            LAZY_DATE_FORMAT = () -> DataTypeUtils.getDateFormat(dateFormat);
+        }
+
+        if (isEmpty(args.getTimeFormat())) {
+            this.timeFormat = null;
+            LAZY_TIME_FORMAT = null;
+        } else {
+            this.timeFormat = args.getTimeFormat();
+            LAZY_TIME_FORMAT = () -> DataTypeUtils.getDateFormat(timeFormat);
+        }
+
+        if (isEmpty(args.getTimestampFormat())) {
+            this.timestampFormat = null;
+            LAZY_TIMESTAMP_FORMAT = null;
+        } else {
+            this.timestampFormat = args.getTimestampFormat();
+            LAZY_TIMESTAMP_FORMAT = () -> DataTypeUtils.getDateFormat(timestampFormat);
+        }
+
+        try {
+            this.rowIterator = new RowIterator(args.getInputStream(), desiredSheets, args.getFirstRow(), logger);
+        } catch (RuntimeException e) {
+            String msg = "Error occurred while processing record file";
+            logger.error(msg, e);
```

Comment:
It is not necessary to log and error and throw the exception in this case. Recommend removing the logger.error call and allowing the framework to log the exception.